tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -265,17 +265,43 @@ namespace tcf {
             // Convert workload_id from hex string to string
             ByteArray workload_bytes = HexStringToBinary(workload_id);
             std::string workload_type(workload_bytes.begin(), workload_bytes.end());
-            WorkloadProcessor *processor = 
-                WorkloadProcessor::CreateWorkloadProcessor(workload_type);
-            tcf::error::ThrowIf<tcf::error::WorkloadError>(
-                processor == nullptr, "Invalid workload id");
-            processor->ProcessWorkOrder(
-                        workload_type,
-                        StrToByteArray(requester_id),
-                        StrToByteArray(worker_id),
-                        StrToByteArray(work_order_id),
-                        in_wo_data,
-                        out_wo_data);
+            if (this->enclave_type == KME_ENCLAVE) {
+                // Create work order data vector having index and
+                // decrypted work order keys
+                std::vector<tcf::WorkOrderData> in_wo_keys;
+                std::vector<tcf::WorkOrderData> out_wo_keys;
+                for (auto d : data_items_in) {
+                    in_wo_keys.emplace_back(
+                        d.workorder_data.index, d.GetDataEncryptionKey());
+                }
+
+                for (auto d : data_items_out) {
+                    out_wo_keys.emplace_back(
+                        d.workorder_data.index, d.GetDataEncryptionKey());
+                }
+
```

Comment:
Create a function and move above logic there inorder to reduce complexity of this function

Reply:
Done